### PR TITLE
Feat/swc loaderfile  Adding SWC loader to load SWC file 

### PR DESCRIPTION
**Pull request by AdityaBITMESRA**
This pull request adds a new module for loading and parsing SWC files

### CODE REVIEW

**Comment by sanjayankur31** [comment]
Converted to a draft @AdityaBITMESRA. We'll mark it as "ready for review" once the various bits have come in.

**Comment by sanjayankur31** [comment]
- add docstrings
- docstrings should contain references to information on SWC standard and so on
- later: consider using something like networkx so that we take advantage of efficient under the hood representations: https://networkx.org/documentation/stable/#

**Comment by sanjayankur31** [comment]
- add unit tests cases for each method

File: pyneuroml/swc/LoadSWC.py
Comment (sanjayankur31) [inline pyneuroml/swc/LoadSWC.py:279]: I'd make this a logger.error or logger.warning perhaps, and we need to decide if this should stop the parsing of the file.

**Comment by sanjayankur31** [comment]
Looking good! Notes from the meeting:

- let's leave the optimisation (networkx etc. bits) for the moment
- rather, let's focus on implementing the functions we'll need so that we have a "public API": `get_parent`, `get_children`, `get_branch_points`, and so on---a bunch of "getters"
- lets test these out and confirm that they work correctly
- later, we can optimise these under the hood

**Comment by AdityaBITMESRA** [comment]
Thankyou for the instructions

**Comment by AdityaBITMESRA** [comment]
Hi @sanjayankur31 , can you have a look to see if I am going correctly?

**Comment by sanjayankur31** [comment]
Sure, will try to do it today, otherwise it may have to wait for Friday---we have a full day conf/symposium tomorrow so wont' be able to get a lot of time at my desk

File: pyneuroml/swc/LoadSWC.py
Comment (sanjayankur31) [inline pyneuroml/swc/LoadSWC.py:226]: we probably want to return a dict with the key being the type and the value being the list of nodes of that type? 

The use case here is "give me branch points in axons and dendrites" and we'll probably want to treat them differently. If not, the user can always iterate over all the values.

File: pyneuroml/swc/LoadSWC.py
Comment (sanjayankur31) [inline pyneuroml/swc/LoadSWC.py:229]: don't forget the type hints and doc string :)

File: pyneuroml/swc/LoadSWC.py
Comment (sanjayankur31) [reply pyneuroml/swc/LoadSWC.py:253]: this still needs to be updated @AdityaBITMESRA

**Review by sanjayankur31 — CHANGES_REQUESTED** [review]
Looking very good @AdityaBITMESRA. It's almost there. I've left a few comments. 

In general, please do remember to use type hints and add docstrings everywhere---they're very very important and we consider them compulsory in our code.

**Comment by AdityaBITMESRA** [comment]
Hi @sanjayankur31. I have addressed all the comments. Please take a look when time permits

**Comment by sanjayankur31** [comment]
Cool, I'll try to do that today. Could you check and fix the issue ruff is pointing out? (If you set up pre-commit, ruff should run on your system and fix things each time you commit---see the contributing.md file)

**Comment by AdityaBITMESRA** [comment]
I have already setup pre-commit hooks but I think the ruff is pointing out to an error in PlotMorphologyVispy.py file(unused imports) which is not the file I  was working on. So what should I do?

**Comment by sanjayankur31** [comment]
Hrm, that must've been fixed in develop. Let me update your branch here on GitHub and that should fix it.

**Comment by sanjayankur31** [comment]
OK, that fixes it. Do remember to pull on your local machine so you get the updates to your branch too.

**Comment by AdityaBITMESRA** [comment]
sure...thankyou

**Comment by sanjayankur31** [comment]
Looks good. I've made some tweaks in #400. Will merge that once the tests pass, and that should merge this one too.

**Pull request merged**
